dashboard/app/handler.go

```diff
@@ -78,6 +79,37 @@ func handleContext(fn contextHandler) http.Handler {
 	})
 }
 
+func isRobot(r *http.Request) bool {
+	userAgent := strings.ToLower(strings.Join(r.Header["User-Agent"], " "))
+	if strings.HasPrefix(userAgent, "curl") ||
```

Comment:
This looks unreliable. I afraid this will add pauses to workflows of more innocent users, but won't do anything for malicious users.

Reply:
Avg. latency for /bug requests is 500.382 ms now.
The throttling rate @a-nogikh used was 1 request/second.

Interactive users will observe 1s delay instead of 500ms.
The scripts will not see 429(throttled).
Malicious users will get the rate 1qps per IP. Scaling the IP count they'll be able to bypass this limitation. But are they the problem for us now?

To eliminate the side-effects for interactive users we can /2 our throttling rate and limit it to 2 qps.

Reply:
I means looking at "User-Agent". It can spoofed in any client. For curl/wget it should be just a command line flag.

Reply:
Ah, I see. This is complementary to the existing throttling, rather than replace it. Makes sense.